kasa/interfaces/energy.py

```diff
@@ -0,0 +1,176 @@
+"""Module for base energy module."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from warnings import warn
+
+from ..emeterstatus import EmeterStatus
+from ..feature import Feature
+from ..module import Module
+
+
+class Energy(Module, ABC):
+    """Base interface to represent a LED module."""
+
+    def _initialize_features(self):
+        """Initialize features."""
+        device = self._device
+        self._add_feature(
+            Feature(
+                device,
+                name="Current consumption",
+                attribute_getter="current_consumption",
+                container=self,
+                unit="W",
+                id="current_consumption",
+                precision_hint=1,
+                category=Feature.Category.Primary,
+            )
+        )
+        self._add_feature(
+            Feature(
+                device,
+                name="Today's consumption",
+                attribute_getter="consumption_today",
+                container=self,
+                unit="kWh",
```

Comment:
Just a side note, but IOT & SMART report currently using different units, we should consolidate everything to use either kWh or Wh.

Reply:
Now consolidated to use kWh.